### PR TITLE
AB#89873 Update Local Authority workflow

### PR DESCRIPTION
**Pull request by dawid-pawulski**

### CODE REVIEW

File: src/HE.Investment.AHP.WWW/Workflows/SiteWorkflow.cs [hunk 1]
```diff
@@ -41,7 +41,7 @@ public SiteWorkflowState CurrentState(SiteWorkflowState targetState)
             { Name: var x } when x.IsNotProvided() => SiteWorkflowState.Name,
 
             // TODO: #89874  add support for Section106 pages
-            // TODO #89873: add support for Local Authority pages
+            { LocalAuthority: var x } when x.IsNotProvided() => SiteWorkflowState.LocalAuthoritySearch,
```
Comment (Antek012) [inline src/HE.Investment.AHP.WWW/Workflows/SiteWorkflow.cs:44]: please update CurrentState tests

File: src/HE.Investment.AHP.WWW/Workflows/SiteWorkflow.cs [hunk 2]
```diff
@@ -58,7 +58,10 @@ private bool CanBeAccessed(SiteWorkflowState state)
             SiteWorkflowState.Name => true,
 
             // TODO: #89874  add support for Section106 pages
-            // TODO: #89873 add support for Local Authority pages
+            SiteWorkflowState.LocalAuthoritySearch => true,
```
Comment (Antek012) [inline src/HE.Investment.AHP.WWW/Workflows/SiteWorkflow.cs:61]: please update StateCanBeAccessed tests

**Review by Antek012 — APPROVED** [review]

File: src/HE.Investment.AHP.WWW/Workflows/SiteWorkflow.cs (hunk 1 above)
Comment (dawid-pawulski) [reply src/HE.Investment.AHP.WWW/Workflows/SiteWorkflow.cs:44]: done

File: src/HE.Investment.AHP.WWW/Workflows/SiteWorkflow.cs (hunk 2 above)
Comment (dawid-pawulski) [reply src/HE.Investment.AHP.WWW/Workflows/SiteWorkflow.cs:61]: done

**Review by jjankowski87 — APPROVED** [review]

**Pull request merged**